zio-http/src/main/scala/zhttp/http/HttpData.scala

```diff
@@ -54,7 +54,7 @@ sealed trait HttpData[-R, +E] { self =>
    */
   private[zhttp] def unsafeSize: Long = self match {
     case HttpData.Empty               => 0L
-    case HttpData.Text(text, _)       => text.length.toLong
+    case HttpData.Text(text, charSet) => text.getBytes(charSet).size.toLong
     case HttpData.BinaryChunk(data)   => data.size.toLong
```

Comment:
We can drop `Text` and `BinaryChunk`

Reply:
I have kept HttpData.Text in the end 
```
  private[zhttp] def unsafeSize: Long = self match {
    case HttpData.Empty               => 0L
    case HttpData.Complete(data)      => data.readableBytes().toLong
    case HttpData.BinaryStream(_)     => -1L
    case HttpData.Text(text, charset) => text.getBytes(charset).size.toLong
    case _                            => -1L
```
As there is one test failing in EncodeClientParamsSpec
```
      testM("content-length") {
        check(clientParamWithFiniteData) { params =>
          val req = encodeClientParams(HttpVersion.HTTP_1_1, params)
          assert(req.headers().getInt(HttpHeaderNames.CONTENT_LENGTH).toLong)(equalTo(params.data.unsafeSize))
        }
      }
```
Looking into it.

Reply:
Done.